src/middlewared/middlewared/plugins/smb_/sharesec.py

```diff
@@ -42,6 +42,40 @@ class Config:
         'data_type': 'BYTES'
     }
 
+    @private
+    async def validate_share_ace(self, schema, ace, verrors):
+        if sid := ace.get('ae_who_sid'):
+            if not await self.middleware.call('idmap.sid_to_unixid', sid):
+                verrors.add(f'{schema}.sid', f'{sid}: SID is unknown')
+
+            return
```

Comment:
Does this return need to be here or in scope of the verrors?

Reply:
Nah, If we have a valid SID we don't need to check the username because it will be ignored.